pkg/secretlock/env/env_secret_lock.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package env
+
+import (
+	"crypto/cipher"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/tink/go/subtle/random"
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+)
+
+// Lock is a secret lock service responsible for encrypting keys using a master key stored as environment variable
+type Lock struct {
+	aead cipher.AEAD
+}
+
+// NewService creates a new instance of a local secret lock service
+func NewService(keyURI string) (secretlock.Service, error) {
+	mk := os.Getenv("LOCAL_" + strings.ReplaceAll(keyURI, "/", "_"))
+	if mk == "" {
+		return nil, fmt.Errorf("masterKey not set")
+	}
+
+	masterKey, err := base64.URLEncoding.DecodeString(mk)
+	if err != nil {
+		return nil, err
+	}
+
+	aead, err := chacha20poly1305.NewX(masterKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Lock{aead: aead}, nil
+}
+
+// Encrypt req using key in the local secret lock service (keyURI is used for remote locks, it is ignored by this
+// implementation)
```

Comment:
I see keyURI is always being ignored in these functions. Instead it is being passed into the constructor and then ignored.

 Can you elaborate on this design?

Reply:
to support cloud storage (even Android keystore in Java/iOS Keychain in objectiveC), master keys are referenced by a keyURI. This is an implementation of secretlock.Service

for local/env storage, keyURI is meaningless as it is fetched from a local path/env variable.